packages/@lwc/integration-karma/test/component/LightningElement.attachInternals/elementInternals/sanity/index.spec.js

```diff
@@ -0,0 +1,55 @@
+import { createElement } from 'lwc';
+import { ariaProperties, ariaAttributes } from 'test-utils';
+
+import ElementInternal from 'ei/component';
+
+if (process.env.NATIVE_SHADOW && typeof ElementInternals !== 'undefined') {
+    let elm;
+    beforeEach(() => {
+        elm = createElement('ei-component', { is: ElementInternal });
+        document.body.appendChild(elm);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(elm);
+    });
+
+    describe('ElementInternals', () => {
+        it('should be associated to the correct element', () => {
+            // Ensure external and internal views of shadowRoot are the same
+            expect(elm.internals.shadowRoot).toBe(elm.template);
+            expect(elm.internals.shadowRoot).toBe(elm.shadowRoot);
+        });
+
+        // Firefox does not support ARIAMixin inside ElementInternals.
+        // Check to see if ARIAMixin value is defined on ElementInternals before
+        // testing accessibility.
+        if (Object.prototype.hasOwnProperty.call(window.ElementInternals.prototype, 'ariaAtomic')) {
```

Comment:
This [may change](https://bugzilla.mozilla.org/show_bug.cgi?id=1785412) with Firefox 119. Might be worth testing this in Firefox Nightly.

Reply:
You're right it's there in Firefox Nightly! Added a TODO to remove this check once it's officially supported.

Reply:
I'm more worried about Fiefox having partial support and it breaking some of our tests.

Do they pass our tests fully?

Reply:
I'll need to take a deeper look but at first glance it looks like Firefox nightly breaks all the aria polyfill tests.

Reply:
Opened an issue: https://github.com/salesforce/lwc/issues/3753